tests/pytest/test_resp3.py

```diff
@@ -0,0 +1,169 @@
+        
+        r.assertEqual(r.execute_command('JSON.DEL', 'test_resp3', '$..b'), 2)
+        
+        # Test none existing path
+        r.assertEqual(r.execute_command('JSON.DEL', 'test_resp3', '$.a1.b'), 0)
+
+        # Test none existing key
+        r.assertEqual(r.execute_command('JSON.DEL', 'test_no_such_key', '$.a1.b'), 0)
+
+        # Test not a JSON key
+        r.expect('JSON.DEL', 'test_not_JSON', '$.a1.b').raiseError()
+
+    # Test JSON.NUMINCRBY RESP3
+    def test_resp_json_num_ops(self):
+        r = self.env
+
+        r.assertTrue(r.execute_command('SET', 'test_not_JSON', 'test_not_JSON'))
+
+        r.assertOk(r.execute_command('JSON.SET', 'test_resp3', '$', '{"a1":{"b":{"c":1}},"a2":{"b":{"c":2.2}},"a3":{"b":{"c":"val"}}}'))
+
+        # Test NUMINCRBY
+        r.assertEqual(r.execute_command('JSON.NUMINCRBY', 'test_resp3', '$.a1.b.c', 1), [2])
+        r.assertEqual(r.execute_command('JSON.NUMINCRBY', 'test_resp3', '$..c', 2), [4, 4.2, None])
+
+        # Test NUMMULTBY
+        r.assertEqual(r.execute_command('JSON.NUMMULTBY', 'test_resp3', '$.a2.b.c', 2.3), [9.66])
+        r.assertEqual(r.execute_command('JSON.NUMMULTBY', 'test_resp3', '$..c', 2), [8, 19.32, None])
+
+        # Test none existing key
+        r.expect('JSON.NUMINCRBY', 'test_no_such_key', '$.a1.b', '1').raiseError()
+
+        # Test not a JSON key
+        r.expect('JSON.NUMMULTBY', 'test_not_JSON', '$.a1.b', '1').raiseError()
+
+
+    # Test JSON.MSET RESP3
+    def test_resp_json_mset(self):
+        r = self.env
+
+        r.assertTrue(r.execute_command('SET', 'test_not_JSON', 'test_not_JSON'))
+
+        r.assertOk(r.execute_command('JSON.MSET', 'test_resp3_1{s}', '$', '{"a1":{"b":{"c":1}},"a2":{"b":{"c":2}}}', 'test_resp3_2{s}', '$', '{"a1":{"b":{"c":1}},"a2":{"b":{"c":2}}}'))
+
+        # Test none existing key
+        r.expect('JSON.MSET', 'test_no_such_key', '$.a1.b', '1').raiseError()
+
+        # Test not a JSON key
+        r.expect('JSON.MSET', 'test_not_JSON', '$.a1.b', '1').raiseError()
+
+    # Test JSON.MERGE RESP3
+    def test_resp_json_merge(self):
+        r = self.env
+
+        r.assertTrue(r.execute_command('SET', 'test_not_JSON', 'test_not_JSON'))
+
+        r.assertOk(r.execute_command('JSON.SET', 'test_resp3', '$', '{"a1":{"b":{"c":1}},"a2":{"b":{"c":2}}}'))
+
+        r.assertOk(r.execute_command('JSON.MERGE', 'test_resp3', '$', '{"a3":4}'))
+
+        # Test none existing key
+        r.expect('JSON.MERGE', 'test_no_such_key', 'test_resp3_1', '$', '{"a3":4}').raiseError()
+
+        # Test not a JSON key
+        r.expect('JSON.MERGE', 'test_not_JSON', '$', '{"a3":4}').raiseError()
+
+    # Test JSON.TYPE RESP3
+    def test_resp_json_type(self):
+        r = self.env
+            
+        r.assertTrue(r.execute_command('SET', 'test_not_JSON', 'test_not_JSON'))
+
+        r.assertOk(r.execute_command('JSON.SET', 'test_resp3', '$', '{"a1":{"b":{"c":1}},"a2":{"b":{"c":2}}}'))
+
+        # Test JSON.TYPE RESP3
```

Comment:
No coverage for double (number), boolean, null, and array.

Reply:
yes, this test wasn't meant to cover `JSON.TYPE` but only to test the return format.
But, I see that we don't cover them on other tests either so I'll extend it